cub/test/test_device_batch_memcpy.cu

```diff
@@ -252,20 +252,42 @@ void RunTest(BufferOffsetT num_buffers,
     num_total_bytes += h_buffer_sizes[i];
   }
 
-  // Shuffle input buffer source-offsets
-  std::uint_fast32_t shuffle_seed = 320981U;
-  if (input_gen == TestDataGen::RANDOM)
-  {
-    h_buffer_src_offsets = GetShuffledBufferOffsets<BufferOffsetT, ByteOffsetT>(h_buffer_sizes,
-                                                                                shuffle_seed);
-    shuffle_seed += 42;
-  }
+  const bool is_integrated_gpu = IsIntegrated();
 
-  // Shuffle input buffer source-offsets
-  if (output_gen == TestDataGen::RANDOM)
   {
-    h_buffer_dst_offsets = GetShuffledBufferOffsets<BufferOffsetT, ByteOffsetT>(h_buffer_sizes,
-                                                                                shuffle_seed);
+    // Shuffle input buffer source-offsets
+    std::uint_fast32_t shuffle_seed = 320981U;
+
+    const std::size_t total_required_mem = 2 * sizeof(BufferOffsetT) * h_buffer_sizes.size()
+                                         + 2 * sizeof(ByteOffsetT) * h_buffer_sizes.size();
+
+    if (is_integrated_gpu && FreeGlobalMem() < total_required_mem)
```

Comment:
Would it make sense to check available memory in the `is_integrated` case also before [host-side memory allocations on lines 216-219](https://github.com/NVIDIA/cccl/pull/529/files#diff-fd0f543760e5eee4c394daa870af3243357edec50f5780018dd313fcbe617823R216-R219)? If we decided to introduce those checks, we should also consider adding those to `cub/test/test_device_batch_copy.cu`

Reply:
nit: Should we leave a comment that this memory check relates to implementation-specific host-side memory requirements in `GetShuffledBufferOffsets`?